peps/pep-0790.rst

```diff
@@ -257,6 +257,18 @@ Backwards Compatibility
 This does not affect any existing ``pyproject.toml``'s, since this was strictly
 not allowed before this PEP.
 
+``pyproject.toml``'s opting in to have the static fields dynamically extended can have
+backards incompatibilities depending on the build frontend and toolings used to interpret
+the static data:
+* When using pre-computed metadata from the sdist or wheel, this change is backwards
+  compatible.
+* The build backend SHOULD be lower-pinned to the version that supports this extension.
+* The build frontend SHOULD be compatible with this change or use a source with
+  pre-computed metadata such as an sdist.
```

Comment:
I probably have some concepts here mixed up and it needs a bit more refinement. The idea is that `pip`/`uv` + (build-backend?) can work with the `sdist` and the new `pyproject.toml`, but otherwise there is a backwards incompatibility that has to be accounted for in either the build frontend, the build backend of `pyproject-metadata` if that is used as a dependency. It would be a bit tricky to port this support to the downstream distros particularly the LTS, I will try to get some further opinions on this, but hopefully that can be resolved with the latter point.

Reply:
> Build back-ends MUST raise an error if the metadata specifies a field statically as well as being listed in dynamic.

This is the current requirement.  It does not say build frontends should do so.